cfme/tests/automate/custom_button/test_cloud_objects.py

```diff
@@ -49,15 +63,23 @@ def setup_objs(button_group, provider):
         obj = [provider, network_manager]
     elif obj_type == "VM_INSTANCE":
         obj = [provider.appliance.provider_based_collection(provider).all()[0]]
+    elif obj_type == "AZONE":
+        obj = [
+            provider.appliance.collections.cloud_av_zones.filter({"provider": provider}).all()[0]
```

Comment:
Question: Why ``` provider.appliance.collections``` and not ```appliance.collections```. Seems it is the same thing:
```
In[8]: ocp
Out[8]: OpenStackProvider(endpoints={'default': <cfme.infrastructure.provider.openstack_infra.RHOSEndpoint object at 0x7f5962740090>}, name='RHOS11', key='rhos11', zone='default', api_port=5000, api_version='Keystone v3', sec_protocol='Non-SSL', amqp_sec_protocol=None, keystone_v3_domain_id='default', tenant_mapping=False, infra_provider=None)
In[9]: ocp.appliance.collections.cloud_av_zones.all()
Out[9]: 
[AvailabilityZone(name=u'nova', provider=OpenStackProvider(endpoints={'default': <cfme.infrastructure.provider.openstack_infra.RHOSEndpoint object at 0x7f59407a3cd0>}, name='RHOS11', key='rhos11', zone='default', api_port=5000, api_version='Keystone v3', sec_protocol='Non-SSL', amqp_sec_protocol=None, keystone_v3_domain_id='default', tenant_mapping=False, infra_provider=None)),
 AvailabilityZone(name=u'No Availability Zone', provider=OpenStackProvider(endpoints={'default': <cfme.infrastructure.provider.openstack_infra.RHOSEndpoint object at 0x7f59407a3cd0>}, name='RHOS11', key='rhos11', zone='default', api_port=5000, api_version='Keystone v3', sec_protocol='Non-SSL', amqp_sec_protocol=None, keystone_v3_domain_id='default', tenant_mapping=False, infra_provider=None))]
In[10]: appliance.collections.cloud_av_zones.all()
Out[10]: 
[AvailabilityZone(name=u'nova', provider=OpenStackProvider(endpoints={'default': <cfme.infrastructure.provider.openstack_infra.RHOSEndpoint object at 0x7f5940793ed0>}, name='RHOS11', key='rhos11', zone='default', api_port=5000, api_version='Keystone v3', sec_protocol='Non-SSL', amqp_sec_protocol=None, keystone_v3_domain_id='default', tenant_mapping=False, infra_provider=None)),
 AvailabilityZone(name=u'No Availability Zone', provider=OpenStackProvider(endpoints={'default': <cfme.infrastructure.provider.openstack_infra.RHOSEndpoint object at 0x7f5940793ed0>}, name='RHOS11', key='rhos11', zone='default', api_port=5000, api_version='Keystone v3', sec_protocol='Non-SSL', amqp_sec_protocol=None, keystone_v3_domain_id='default', tenant_mapping=False, infra_provider=None))]
```

Reply:
@jawatts I have `provider` object so I am directly using `appliance` attribute of it in a state of inheriting `appliance`. Both are the same but I like to avoid `appliance` inheritance when I have `provider` obj.

Suppose in the same fixture; I need collection only then `fixture` scope I can put as `module` with `appliance` inheritance. But, I need `provider` so no longer `functional` scope so I directly use `provider. appliance`.